File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        int[] dp = new int[n+1];
        int p1=1, p2=1;
        
        for(int i = 1; i <= n; i++) {
            dp[i] = dp[i-1] + i;                           
        }
        
        
        while(p1 <= p2 && p2 <= n) {
            int now = dp[p2] - dp[p1] + p1;
            if(now > n) p1++;
            if(now < n) p2++;
            if(now==n) {
                answer++;
                p1++;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    ans = 0
    dp = []
    dp.append(0)
    dp.append(1)
    for i in range(2, n+1):
        dp.append(dp[-1]+i)
        
    p1 = 1
    p2 = 1
    while p1 <= p2 and p2 <= n+1:
        now = dp[p2] - dp[p1-1]
        if now < n: p2 += 1
        if now >= n: 
            if now == n:
                ans += 1
            p1 += 1
        
        
    return ans